### PR TITLE
Scala-Native 0.5.0 support

### PR DESCRIPTION
**Pull request by lihaoyi**
This required a Mill version upgrade to 0.11.x, which involved cleaning up a lot of now-unnecessary boilerplate in the `build.sc`.

Same version changes as elsewhere: drop 2.11.x/3.1.x/3.2.x support, add Scala-Native 0.5.0

### CODE REVIEW

**Comment by lihaoyi** [comment]
Review by @lolgab @lefou

File: build.sc [hunk 1]
```diff
@@ -28,7 +24,7 @@ trait SourcecodeModule extends PublishModule with MimaCheck {
   // Temporary until the next version of Mima gets released with
   // https://github.com/lightbend/mima/issues/693 included in the release.
   def mimaPreviousArtifacts =
-    if(isScala3(crossScalaVersion)) Agg.empty[Dep] else super.mimaPreviousArtifacts()
+    if(crossScalaVersion.startsWith("3.")) Agg.empty[Dep] else super.mimaPreviousArtifacts()
```
Comment (lolgab) [inline build.sc:27]: `isScala3` still exists but is inside `ZincWorkerUtil` now I believe

File: build.sc [hunk 2]
```diff
@@ -41,98 +37,42 @@ trait SourcecodeModule extends PublishModule with MimaCheck {
...
-      def scalaNativeVersion = crossScalaNativeVersion
-      def offset = os.up
-      def moduleDeps = Seq(NativeSourcecodeModule.this)
-      val crossScalaVersion = NativeSourcecodeModule.this.crossScalaVersion
+  object native extends Cross[NativeSourcecodeModule](scalaVersions)
+  trait NativeSourcecodeModule extends SourcecodeMainModule with ScalaNativeModule with SourcecodeModule {
+
+    def scalaNativeVersion = "0.5.0"
+    object test extends ScalaNativeTests{
+      // stub to make use of test plumbing but not running a test suite
```
Comment (lolgab) [inline build.sc:73]: Why can't we run tests on Scala Native?

Comment (lihaoyi) [reply build.sc:73]: @lolgab sourcecode doesnt use a test framework, it just runs through the asserts starting from `def main`. But the build level plumbing for nested test modules w.r.t. cross version and cross platform modules is very handy, so I'm abusing it a bit

File: build.sc (hunk 1 above)
Comment (lihaoyi) [reply build.sc]: fixed

**Pull request merged**